drivers/i3c/i3c_npcx.c

```diff
@@ -0,0 +1,2055 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/*132*/
+#define DT_DRV_COMPAT nuvoton_npcx_i3c
+
+#include <string.h>
+
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/sys_io.h>
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/i3c.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/reset.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(npcx_i3c, CONFIG_I3C_LOG_LEVEL);
+
+#define NPCX_I3C_CHK_TIMEOUT 10000    /* Unit: us, timeout for checking register status */
+#define I3C_SCL_PP_FREQ_MAX  12500000 /* Unit: Mhz*/
+#define I3C_SCL_OD_FREQ_MAX  4170000  /* Unit: Mhz*/
+
+#define I3C_BUS_TLOW_PP_MIN_NS  24  /* T_LOW period in push-pull mode */
+#define I3C_BUS_THigh_PP_MIN_NS 24  /* T_High period in push-pull mode */
+#define I3C_BUS_TLOW_OD_MIN_NS  200 /* T_LOW period in open-drain mode */
+
+#define PPBAUD_DIV_MAX (BIT(GET_FIELD_SZ(NPCX_I3C_MCONFIG_PPBAUD)) - 1) /* PPBAUD divider max */
+
+#define DAA_TGT_INFO_SZ 0x8 /* 8 bytes = PID(6) + BCR(1) + DCR(1) */
+#define BAMATCH_DIV     0x4 /* BAMATCH = APB4_CLK divided by four */
+
+/* Default maximum time we allow for an I3C transfer (unit:ms) */
+#define I3C_TRANS_TIMEOUT K_MSEC(100)
```

Comment:
I would spec this in MS (`I3C_TRANS_TIMEOUT_MS`) and use `K_MSEC` in the usage, just cor coherency with how other time units in the file are declared